packages/@tinacms/graphql/src/database/index.ts

```diff
@@ -73,7 +74,93 @@ export type CreateDatabase = {
   version?: boolean
 }
 
+export interface GitProvider {
+  onPut: (key: string, value: string) => Promise<void>
+  onDelete: (key: string) => Promise<void>
+}
+
+export type CreateDatabase = Omit<
+  DatabaseArgs,
+  'level' | 'onPut' | 'onDelete'
+> & {
+  databaseAdapter: Level
+  gitProvider: GitProvider
+
+  /**
+   * @deprecated Use databaseAdapter instead
+   */
+  level?: Level
+  /**
+   * @deprecated Use gitProvider instead
+   */
+  onPut?: OnPutCallback
+  /**
+   * @deprecated Use gitProvider instead
+   */
+  onDelete?: OnDeleteCallback
+}
+
+export type CreateLocalDatabaseArgs = Omit<DatabaseArgs, 'level'> & {
+  port?: number
+  rootPath?: string
+}
+
+export const createLocalDatabase = (config?: CreateLocalDatabaseArgs) => {
+  const level = new TinaLevelClient(config?.port)
+  level.openConnection()
+  const fsBridge = new FilesystemBridge(config?.rootPath || process.cwd())
+  return new Database({
+    bridge: fsBridge,
+    ...(config || {}),
+    level,
+  })
+}
+
 export const createDatabase = (config: CreateDatabase) => {
+  if (config.onPut && config.onDelete) {
+    console.warn(
+      'onPut and onDelete are deprecated. Please use gitProvider.onPut and gitProvider.onDelete instead.'
+    )
+  }
+  if (config.level) {
+    console.warn('level is deprecated. Please use databaseAdapter instead.')
+  }
+  if (
+    config.onPut &&
+    config.onDelete &&
+    config.level &&
+    !config.databaseAdapter &&
+    !config.gitProvider
+  ) {
+    // This is required for backwards compatibility
+    return new Database({
+      ...config,
+      level: config.level,
+    })
+  }
+
+  if (!config.gitProvider) {
+    throw new Error(
+      'createDatabase requires a gitProvider. Please provide a gitProvider.'
+    )
+  }
```

Comment:
I discovered that createDatabase is blowing up in tina cloud whenever gitProvider is not specified even when a bridge is passed in. I don't think gitProvider should be required if bridge is specified.